jax/experimental/jax2tf/tests/savedmodel_test.py

```diff
@@ -100,6 +101,25 @@ def f_jax_jvp(primals, tangents):
     with self.assertRaisesRegex(TypeError, "An op outside of the function building code is being passed"):
       _ = tape.gradient(y, xv)
 
+  def test_xla_context_preserved(self):
+    # Certain ops are converted to ensure an XLA context, e.g.,
+    # tf.gather, so that the index-out-of-bounds behavior matches that of
+    # JAX. We check that this information is preserved through a savedmodel
+    nr_elements = 10
+    def f_jax(arr):
+      return lax.dynamic_slice(arr, [nr_elements + 1], [1])  # out of bounds, should return the last element
+    arr = np.arange(nr_elements, dtype=np.float32)
+    self.assertEqual(nr_elements - 1, f_jax(arr))
+
+    f_tf = jax2tf.convert(f_jax)  # Even in eager evaluation we get XLA behavior
+    self.assertEqual(nr_elements - 1, f_tf(arr))
+
+    model = tf.Module()
+    model.f = tf.function(jax2tf.convert(f_jax),
+                          autograph=False,
+                          input_signature=[tf.TensorSpec([nr_elements], tf.float32)])
+    restored_model = self.save_and_load_model(model)
+    self.assertEqual(nr_elements - 1, restored_model.f(arr))
```

Comment:
Should we add tests for gather/scatter as well?

Reply:
We could, but it should not be necessary, because I want to test that the _xla_compile mechanism works through savedmodel. This is just checking that the savedmodel encoded the tags that require XLA for certain ops. This should not depend on the actual op.